webhook-service/src/main/scala/io/renku/webhookservice/eventstatus/StatusInfo.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2022 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.webhookservice.eventstatus
+
+import cats.syntax.all._
+import io.circe.Encoder
+import io.circe.literal._
+import io.renku.graph.model.events.{EventStatus, EventStatusProgress}
+import EventStatus._
+
+private sealed trait StatusInfo {
+  val activated: Boolean
+  val progress:  Progress
+}
+
+private object StatusInfo {
+
+  final case class ActivatedProject(progress: Progress.NonZero, details: Details) extends StatusInfo {
+    override val activated: Boolean = true
+  }
+
+  def activated(eventStatus: EventStatus): StatusInfo.ActivatedProject =
+    ActivatedProject(Progress.from(eventStatus), Details(eventStatus))
+
+  final case object NotActivated extends StatusInfo {
+    override val activated: Boolean  = false
+    override val progress:  Progress = Progress.Zero
+  }
+
+  implicit def encoder[PS <: StatusInfo]: Encoder[PS] = {
+    case info @ ActivatedProject(status: Progress.NonZero, details) => json"""{
+      "activated": ${info.activated},
+      "progress": {
+        "done":       ${status.statusProgress.stage.value},
+        "total":      ${EventStatusProgress.Stage.Final.value},
+        "percentage": ${status.statusProgress.completion.value}
+      },
+      "details": {
+        "status":  ${details.status},
+        "message": ${details.message}
+      }
+    }"""
+    case info @ NotActivated => json"""{
+      "activated": ${info.activated},
+      "progress": {
+         "done":       0,
+         "total":      ${EventStatusProgress.Stage.Final.value},
+         "percentage": 0.00
+       }
+    }"""
+  }
+}
+
+private sealed trait Progress extends Product with Serializable
+
+private object Progress {
+
+  final case object Zero extends Progress {
+    lazy val finalStage: EventStatusProgress.Stage = EventStatusProgress.Stage.Final
```

Comment:
I probably just don't see it in the PR, I was wondering why this field to a constant is necessary? Same in `NonZero`. It's not in the interface and always points to the same constant value.

Reply:
Yes, good point